ste/xfer-setProperties.go

```diff
@@ -0,0 +1,224 @@
+package ste
+
+import (
+	"fmt"
+	"github.com/Azure/azure-pipeline-go/pipeline"
+	"github.com/Azure/azure-storage-azcopy/v10/azbfs"
+	"github.com/Azure/azure-storage-azcopy/v10/common"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/Azure/azure-storage-file-go/azfile"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+//var explainedSkippedRemoveOnce sync.Once
+
+func getBlockBlobTierString(tier common.BlockBlobTier) azblob.AccessTierType {
+	switch tier {
+	case common.EBlockBlobTier.Hot():
+		return azblob.AccessTierHot
+	case common.EBlockBlobTier.Cool():
+		return azblob.AccessTierCool
+	case common.EBlockBlobTier.Archive():
+		return azblob.AccessTierArchive
+	case common.EBlockBlobTier.None():
+		return azblob.AccessTierNone
+	default:
+		panic("invalid tier type")
+	}
+}
+
+func getPageBlobTierString(tier common.PageBlobTier) azblob.AccessTierType {
+	switch tier {
+	case common.EPageBlobTier.P10():
+		return azblob.AccessTierP10
+	case common.EPageBlobTier.P15():
+		return azblob.AccessTierP15
+	case common.EPageBlobTier.P20():
+		return azblob.AccessTierP20
+	case common.EPageBlobTier.P30():
+		return azblob.AccessTierP30
+	case common.EPageBlobTier.P4():
+		return azblob.AccessTierP4
+	case common.EPageBlobTier.P40():
+		return azblob.AccessTierP40
+	case common.EPageBlobTier.P50():
+		return azblob.AccessTierP50
+	case common.EPageBlobTier.P6():
+		return azblob.AccessTierP6
+	case common.EPageBlobTier.None():
+		return azblob.AccessTierNone
+	default:
+		panic("Invalid tier type")
+	}
+}
+func SetProperties(jptm IJobPartTransferMgr, p pipeline.Pipeline, pacer pacer) {
+	// If the transfer was cancelled, then reporting transfer as done and increasing the bytes transferred by the size of the source.
+	if jptm.WasCanceled() {
+		jptm.ReportTransferDone()
+		return
+	}
+
+	// schedule the work as a chunk, so it will run on the main goroutine pool, instead of the
+	// smaller "transfer initiation pool", where this code runs.
+	id := common.NewChunkID(jptm.Info().Source, 0, 0)
+	cf := createChunkFunc(true, jptm, id, func() {
+		to := jptm.FromTo()
+		switch to.From() {
```

Comment:
Nitpicking: Generally speaking, our individual chunk functions don't operate across multiple different endpoints. This could be a function that returns a func() that has a single, dedicated purpose.

Reply:
@mohsha-msft can you help me with this?